### ISSUE TITLE
Argu 5.1.0 update causes MissingMethodException

### ISSUE THREAD

**Issue by drivehappy** [haf/expecto#227]
After updating to the latest version of Argu (5.1.0) I'm receiving the following with Expecto 6.0.0:

`System.MissingMethodException: 'Method not found: 'Void Argu.ArgumentParser`1..ctor(Microsoft.FSharp.Core.FSharpOption`1<System.String>, Microsoft.FSharp.Core.FSharpOption`1<System.String>, Microsoft.FSharp.Core.FSharpOption`1<Int32>, Microsoft.FSharp.Core.FSharpOption`1<Argu.IExiter>)'.'`

**Comment by AnthonyLloyd** [comment]
I think we need to upgrade Argu, we are on 3.7. I will look into this.

I guess you are not using paket which would tell you your dependencies?

**Comment by kellerd** [comment]
Hi guys.

Seems Expecto lists >= 3.7 for Argu, so paket picks the most recent version.  I'm running a netcoreapp2.0

```
paket.exe why Argu
NuGet Argu - 5.1 is a transitive dependency.
It is part of following dependency chains:

-> Expecto - 6.0
  -> Argu - 5.1
```

```
    Argu (5.1) - restriction: || (>= net461) (>= netstandard1.6)
      FSharp.Core (>= 4.0.0.1) - restriction: >= net45
      FSharp.Core (>= 4.3.2) - restriction: && (< net45) (>= netstandard2.0)
      System.Configuration.ConfigurationManager (>= 4.4) - restriction: && (< net45) (>= netstandard2.0)
    Expecto (6.0)
      Argu (>= 3.7) - restriction: || (>= net461) (>= netstandard1.6)
      FSharp.Core (>= 4.1.18) - restriction: || (>= net461) (>= netstandard1.6)
      Mono.Cecil (>= 0.10.0-beta7) - restriction: || (>= net461) (>= netstandard1.6)
      System.Diagnostics.FileVersionInfo (>= 4.3) - restriction: || (&& (< net461) (>= netstandard1.6) (< netstandard2.0)) (&& (< net461) (>= netstandard2.0))
```

**Comment by forki** [comment]
Yes look at paket's build - ilrepack works well for this case

**Comment by haf** [comment]
The dependency should be ~> MAJOR, so that it would have ended up like this: `Argu >= 3.7 && < 4`.

+1 for upgrading.

Be wary of cross-platform issues with ilrepack. I've never gotten it to work cross mono/.net and now there's even .net core to consider. Keep it simple and just release an upgrade.

**Comment by haf** [comment]
@kellerd @forki @drivehappy It would be nice with a PR that is less strict on the Argu dependency :)

@forki Roger that.

**Comment by AnthonyLloyd** [comment]
Argu has dropped netstandard1.6. Does this mean Expecto has to?

**Comment by haf** [comment]
I think it's nicer to have Argu up to date as that will allow us to get its improvements; like env var parsers out of the box and reflected definitions.

I'm still not in favour of ILMerging it as Expecto should be possible to build libs out of; e.g. have a look at https://github.com/haf/canopy/blob/waiting/tests/ParallelUsage/Program.fs#L10 — or would ILMerging allow callers to interact with Argu?

**Comment by AnthonyLloyd** [comment]
Are people tending to drop netstandard1.6 support? I'm not sure if its used.

**Comment by forki** [comment]
nuke it ;-)

**Comment by AnthonyLloyd** [comment]
Another major version! Will release this tonight,

**Comment by AnthonyLloyd** [comment]
Released upgrade to Argu 5.1.0 in Expecto v7.0.0. Will ilrepack if this comes up again in the future.
